GDPR-Phase2-Java/src/main/java/com/gdpr/phase2/encoder/Base64Url.java

```diff
@@ -0,0 +1,50 @@
+package com.gdpr.phase2.encoder;
+
+import org.apache.commons.codec.binary.Base64;
+import org.apache.commons.lang3.StringUtils;
+
+
+import java.util.regex.Pattern;
+
+public class Base64Url {
+
+    private Base64Url() {
+    }
+    private static final Base64Url instance = new Base64Url();
+    public static Base64Url getInstance() {
+        return instance;
+    }
+
+    private static final int ONE_BYTE = 8;
+    private static final Pattern p = Pattern.compile("[A-Z0-9]",Pattern.CASE_INSENSITIVE);
```

Comment:
Base64Url encoding can have `A-Z`, `a-z`, `0-9`, `-`, and `_` characters.

Reply:
So I think your regex pattern should be `Pattern.compile("[A-Z0-9\-_]",Pattern.CASE_INSENSITIVE);`

Reply:
okk